nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -296,62 +303,91 @@ async fn run() -> Result<()> {
                 continue;
             }
 
-            match aggr_book.entry(job_id.clone()) {
-                Entry::Occupied(mut entry) => {
-                    tracing::debug!("Another result for job {} found", job_id);
-                    let (workers, payload) = entry.get_mut();
+            // Falling below are logics for classification
+            let raw_data_id = hex::encode(new_payload.header.raw_data_id.hash());
+            let mut fetch_connection = redis_client.get_connection();
+            if fetch_connection.is_err() {
+                tracing::debug!("Failed to fetch redis connection, abort");
+                continue;
+            }
 
-                    if *payload != new_payload {
-                        // ToDo: here the logic can be complicated. For example, how do we
-                        //  handle resolutions now? Do we need a separate book to handle
-                        //  controversies?
-                        tracing::error!("payloads for {} are different", job_id);
-                        RESULT_ERRORS.inc();
+            // First fetch all existing tasks
+            let saved_tasks: Result<String, RedisError> =
+                (&mut fetch_connection.unwrap()).get(raw_data_id.clone());
+            let mut tasks: Vec<AssignerTask> = match saved_tasks {
+                Ok(val) => {
+                    match from_str(val.as_str()) {
+                        Ok (t) => {
+                            t
+                        }
 
-                        // ToDo: Currently we handle disputes by deleting the entry altogether.
-                        //  If this was already at the second result, another one will come and
-                        //  linger in the book because no other matches are expected to come.
-                        //  Another approach needs to decided.
-                        entry.remove();
-                        PENDING_JOBS.set(aggr_book.len()as i64);
-                        continue;
+                        Err(err) => {
+                            tracing::debug!("Failed to fetch tasks from redis for data: {}", raw_data_id.clone());
+                            Vec::new()
+                        }
                     }
+                }
+                Err(_) => {
+                    PENDING_JOBS.inc();
```

Comment:
Why do we update pending jobs if we cannot retrieve it? We don't know, maybe it's not possible to retrieve it 10 times, and we are counting a PENDING job 10 times